### PR TITLE
Jwoo/ios select actions

### PR DESCRIPTION
**Pull request by nesalang**
- Implemented Select Actions for image/container/column set/column

### CODE REVIEW

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRShowCardTarget.mm [hunk 1]
```diff
@@ -35,16 +36,103 @@ - (instancetype)initWithAdaptiveCard:(std::shared_ptr<AdaptiveCards::AdaptiveCar
...
+                    green:((num & 0x0000FF00) >>  8) / 255.0
+                     blue:((num & 0x000000FF)) / 255.0
+                    alpha:((num & 0xFF000000) >> 24) / 255.0];
+    [wrappingView setAlignmentForSubview:AdaptiveCards::HorizontalAlignment::Center];
+    [_superview addArrangedSubview:_adcView];
+}
+
+- (IBAction)showCard
+{
+    if(YES == _isItCurrentlyShown)
```
Comment (bviglietta) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRShowCardTarget.mm]: >_isItCurrentlyShown [](start = 14, length = 19)

It looks like `_isItCurrentlyShown` is equivalent to `_adcView && !_adcView.hidden`. If that's right, it's better to remove the redundant field.

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRShowCardTarget.mm [hunk 2]
```diff
@@ -35,16 +36,103 @@ - (instancetype)initWithAdaptiveCard:(std::shared_ptr<AdaptiveCards::AdaptiveCar
...
+    wrappingView.backgroundColor =
+    [UIColor colorWithRed:((num & 0x00FF0000) >> 16) / 255.0
+                    green:((num & 0x0000FF00) >>  8) / 255.0
+                     blue:((num & 0x000000FF)) / 255.0
+                    alpha:((num & 0xFF000000) >> 24) / 255.0];
+    [wrappingView setAlignmentForSubview:AdaptiveCards::HorizontalAlignment::Center];
+    [_superview addArrangedSubview:_adcView];
+}
+
+- (IBAction)showCard
```
Comment (bviglietta) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRShowCardTarget.mm]: >showCard [](start = 12, length = 8)

`showCard` is a poor name for a method that either shows or hides the card. It's more like "toggle card visibility".

**Comment by bviglietta** [comment]
I'm happy to continue reviewing these changes, but I think they should also be reviewed by someone more knowledgeable about Adaptive Cards. My ability to review is limited by the fact that I don't understand what any of the code is really supposed to do.

My signoff here indicates that everything looks OK to the best of my ability to understand it.

**Comment by nesalang** [comment]
Thanks Ben for reviewing my changes. Yes, I will add someone from our team to review it to address your concerns. 

---
In reply to: [357007192](https://github.com/Microsoft/AdaptiveCards/pull/1067#issuecomment-357007192) [](ancestors = 357007192)

**Pull request merged**